spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/BlasWrapper.scala

```diff
@@ -69,10 +84,82 @@ private[bigdl] class BlasWrapper(val module: AbstractModule[Activity, Activity,
     }).toArray
   }
 
+  /**
+   * Blas layers may not have good performance with mkldnn backend,
+   * so we can use java multi thread to run blas layers and thread number is
+   * related with batch size and core number.
+   * input and output for this module must be in batch.
+   */
+  private def setMultiThreadEnv(input: Activity): Unit = {
+    initEnv = true
+    val multiThread = System.getProperty("multiThread", "false").toBoolean
+    if (this.train && multiThread) {
+      throw new IllegalArgumentException("Please not set multiThread to true for model training")
```

Comment:
what's will be the issue if we set it in training ?

Reply:
Setting mkl omp thread to 1 may be not suitable for training. We need more tests to verify it.